background/handle-site-access.js

```diff
@@ -0,0 +1,24 @@
+chrome.permissions.contains(
+  {
+    permissions: ["webRequest", "webRequestBlocking", "cookies"],
+    origins: ["https://scratch.mit.edu/*", "https://api.scratch.mit.edu/*", "https://clouddata.scratch.mit.edu/*"],
+  },
+  async (hasPermissions) => {
+    if (hasPermissions) {
+      // Load remaining scripts
+      await import("./handle-fetch.js");
+      await import("./handle-cookie.js");
+      await import("./handle-auth.js");
+      await import("./handle-messages.js");
+
+      await import("./get-userscripts.js");
+      await import("./get-persistent-scripts.js");
+    } else {
+      console.error("Could not load addons; site access was not granted");
+      chrome.tabs.create({
+        active: true,
+        url: "/webpages/settings/permissions.html",
+      });
```

Comment:
You could register an event on here so that if the user clicks the Scratch Addons icon, it opens this page as well.
We should also likely redirect to this page if the user goes directly to /settings through the context menu.